Tests fail due to pandas' deprecation warning

Fails with:
```
_______ ERROR collecting tests/unit/bokeh/core/property/test_aliases.py ________
tests/unit/bokeh/core/property/test_aliases.py:24: in <module>
    import pandas as pd
/usr/share/miniconda3/envs/bk-test/lib/python3.12/site-packages/pandas/__init__.py:221: in <module>
    warnings.warn(
E   DeprecationWarning: 
E   Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
E   (to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
E   but was not found to be installed on your system.
E   If this would cause problems for you,
E   please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466
```
I'm guessing we should add pyarrow as a dependency now? Or maybe we should just ignore this with `filterwarnings`.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -107,6 +107,7 @@ filterwarnings = [
     '''ignore:numba is not installed\. This example will be painfully slow\.:UserWarning''',
     '''ignore:Could not infer format, so each element will be parsed individually, falling back to `dateutil`\. To ensure parsing is consistent and as-expected, please specify a format\.:UserWarning''',
     '''ignore:datetime\.datetime\.utcfromtimestamp\(\) is deprecated and scheduled for removal in a future version:DeprecationWarning:dateutil''',
+    '''ignore:\s*Pyarrow will become a required dependency of pandas:DeprecationWarning''',
 ]
 
 [tool.coverage.run]

--- a/tests/unit/bokeh/models/test_sources.py
+++ b/tests/unit/bokeh/models/test_sources.py
@@ -399,7 +399,7 @@ Lime,Green,99,$0.39
 
     def test__stream_good_df_with_date_index_data(self) -> None:
         df = pd.DataFrame(
-            index=pd.date_range('now', periods=30, freq='T'),
+            index=pd.date_range('now', periods=30, freq='min'),
             columns=['A'],
             data=np.cumsum(np.random.standard_normal(30), axis=0),
         )
@@ -423,7 +423,7 @@ Lime,Green,99,$0.39
 
     def test__stream_good_dict_of_index_and_series_data(self) -> None:
         df = pd.DataFrame(
-            index=pd.date_range('now', periods=30, freq='T'),
+            index=pd.date_range('now', periods=30, freq='min'),
             columns=['A'],
             data=np.cumsum(np.random.standard_normal(30), axis=0),
         )
@@ -447,7 +447,7 @@ Lime,Green,99,$0.39
 
     def test__stream_good_dict_of_index_and_series_data_transformed(self) -> None:
         df = pd.DataFrame(
-            index=pd.date_range('now', periods=30, freq='T'),
+            index=pd.date_range('now', periods=30, freq='min'),
             columns=['A'],
             data=np.cumsum(np.random.standard_normal(30), axis=0),
         )

--- a/tests/unit/bokeh/util/test_util__serialization.py
+++ b/tests/unit/bokeh/util/test_util__serialization.py
@@ -35,6 +35,8 @@ import bokeh.util.serialization as bus # isort:skip
 # Setup
 #-----------------------------------------------------------------------------
 
+pandas_1x = pd.__version__.startswith("1")
+
 #-----------------------------------------------------------------------------
 # General API
 #-----------------------------------------------------------------------------
@@ -114,7 +116,7 @@ def test_is_datetime_type_non_pandas_types() -> None:
 
 def test_is_datetime_type_pandas_types() -> None:
     assert bus.is_datetime_type(pd.Timestamp(3000000))
-    assert bus.is_datetime_type(pd.Period('1900', 'Y-DEC'))
+    assert bus.is_datetime_type(pd.Period('1900', 'A-DEC' if pandas_1x else 'Y-DEC'))
     assert bus.is_datetime_type(pd.NaT)
 
 def test_convert_datetime_type_non_pandas_types() -> None:
@@ -127,8 +129,8 @@ def test_convert_datetime_type_non_pandas_types() -> None:
 
 def test_convert_datetime_type_pandas_types() -> None:
     assert bus.convert_datetime_type(pd.Timestamp(3000000)) == 3.0
-    assert bus.convert_datetime_type(pd.Period('1900', 'Y-DEC')) == -2208988800000.0
-    assert bus.convert_datetime_type(pd.Period('1900', 'Y-DEC')) == bus.convert_datetime_type(np.datetime64("1900-01-01"))
+    assert bus.convert_datetime_type(pd.Period('1900', 'A-DEC' if pandas_1x else 'Y-DEC')) == -2208988800000.0
+    assert bus.convert_datetime_type(pd.Period('1900', 'A-DEC' if pandas_1x else 'Y-DEC')) == bus.convert_datetime_type(np.datetime64("1900-01-01"))
     assert np.isnan(bus.convert_datetime_type(pd.NaT))
 
 def test_convert_datetime_type_array_ignores_non_datetime_array() -> None:

--- a/tests/unit/bokeh/util/test_util__serialization.py
+++ b/tests/unit/bokeh/util/test_util__serialization.py
@@ -114,7 +114,7 @@ def test_is_datetime_type_non_pandas_types() -> None:
 
 def test_is_datetime_type_pandas_types() -> None:
     assert bus.is_datetime_type(pd.Timestamp(3000000))
-    assert bus.is_datetime_type(pd.Period('1900', 'A-DEC'))
+    assert bus.is_datetime_type(pd.Period('1900', 'Y-DEC'))
     assert bus.is_datetime_type(pd.NaT)
 
 def test_convert_datetime_type_non_pandas_types() -> None:
@@ -127,8 +127,8 @@ def test_convert_datetime_type_non_pandas_types() -> None:
 
 def test_convert_datetime_type_pandas_types() -> None:
     assert bus.convert_datetime_type(pd.Timestamp(3000000)) == 3.0
-    assert bus.convert_datetime_type(pd.Period('1900', 'A-DEC')) == -2208988800000.0
-    assert bus.convert_datetime_type(pd.Period('1900', 'A-DEC')) == bus.convert_datetime_type(np.datetime64("1900-01-01"))
+    assert bus.convert_datetime_type(pd.Period('1900', 'Y-DEC')) == -2208988800000.0
+    assert bus.convert_datetime_type(pd.Period('1900', 'Y-DEC')) == bus.convert_datetime_type(np.datetime64("1900-01-01"))
     assert np.isnan(bus.convert_datetime_type(pd.NaT))
 
 def test_convert_datetime_type_array_ignores_non_datetime_array() -> None:

